3.5-windowsservercore-1709/runtime/Dockerfile

```diff
@@ -1,11 +1,37 @@
+# escape=`
+
 FROM microsoft/windowsservercore:1709
 
-ADD install /build
-RUN DISM /Online /Add-Package /PackagePath:C:\build\microsoft-windows-netfx3-ondemand-package.cab & \
-    DISM /Online /Add-Package /PackagePath:C:\build\patch\Windows10.0-KB3213986-x64.cab  & \
-    rd /s /q C:\build
-RUN set COMPLUS_NGenProtectedProcess_FeatureEnabled=0 & \
-    \Windows\Microsoft.NET\Framework64\v4.0.30319\ngen update & \
-    \Windows\Microsoft.NET\Framework\v4.0.30319\ngen update & \
-    \Windows\Microsoft.NET\Framework64\v2.0.50727\ngen update & \
+# Install .NET Fx 3.5
+RUN powershell -Command `
+        $ErrorActionPreference = 'Stop'; `
+        $ProgressPreference = 'SilentlyContinue'; `
+        Invoke-WebRequest `
+            -UseBasicParsing `
+            -Uri "https://dotnetbinaries.blob.core.windows.net/dockerassets/microsoft-windows-netfx3-1709.zip" `
+            -OutFile microsoft-windows-netfx3.zip; `
+        Expand-Archive microsoft-windows-netfx3.zip; `
+        Remove-Item -Force microsoft-windows-netfx3.zip; `
+        Add-WindowsPackage -Online -PackagePath .\microsoft-windows-netfx3\microsoft-windows-netfx3-ondemand-package.cab; `
+        Remove-Item -Force -Recurse microsoft-windows-netfx3
+
+# Apply latest patch
+RUN powershell -Command `
```

Comment:
Why download the file in the `RUN` directive as opposed to directly using the `ADD` directive?

Reply:
See https://docs.docker.com/engine/userguide/eng-image/dockerfile_best-practices/#add-or-copy
>Because image size matters, using ADD to fetch packages from remote URLs is strongly discouraged; you should use curl or wget instead. That way you can delete the files you no longer need after they’ve been extracted and you won’t have to add another layer in your image.